### PR TITLE
python312Packages.orbax-checkpoint: 0.6.4 -> 0.10.2

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

Changelog: https://github.com/google/orbax/releases/tag/v0.10.2

Blocked by `jax` update (#318995).

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by dotlambda — COMMENTED** [review]
It would be nicer if `meta.changelog` were `https://github.com/google/orbax/blob/${src.rev}/checkpoint/CHANGELOG.md`.

**Comment by GaetanLepage** [comment]
> It would be nicer if `meta.changelog` were `https://github.com/google/orbax/blob/${src.rev}/checkpoint/CHANGELOG.md`.

Sure. I have used `v${version}` because `rev` contains the `refs/tags` prefix.

File: pkgs/development/python-modules/orbax-checkpoint/default.nix [hunk 1]
```diff
@@ -90,7 +90,7 @@ buildPythonPackage rec {
 
   meta = {
     description = "Orbax provides common utility libraries for JAX users";
-    homepage = "https://github.com/google/orbax/tree/main/checkpoint";
+    homepage = "https://github.com/google/orbax/blob/v${version}/checkpoint/CHANGELOG.md";
     changelog = "https://github.com/google/orbax/releases/tag/v${version}";
```
Comment (dotlambda) [inline pkgs/development/python-modules/orbax-checkpoint/default.nix:94]:
```suggestion
    homepage = "https://github.com/google/orbax/tree/main/checkpoint";
    changelog = "https://github.com/google/orbax/blob/${src.rev}/checkpoint/CHANGELOG.md";
```
or use `lib.removePrefix "refs/tags/" src.rev`

Comment (GaetanLepage) [reply pkgs/development/python-modules/orbax-checkpoint/default.nix]: Done

File: pkgs/development/python-modules/orbax-checkpoint/default.nix [hunk 2]
```diff
@@ -91,7 +91,7 @@ buildPythonPackage rec {
   meta = {
     description = "Orbax provides common utility libraries for JAX users";
     homepage = "https://github.com/google/orbax/tree/main/checkpoint";
-    changelog = "https://github.com/google/orbax/releases/tag/v${version}";
+    changelog = "https://github.com/google/orbax/blob/${lib.removePrefix "refs/tags" src.rev}/checkpoint/CHANGELOG.md";
```
Comment (dotlambda) [inline pkgs/development/python-modules/orbax-checkpoint/default.nix:94]:
```suggestion
    changelog = "https://github.com/google/orbax/blob/${lib.removePrefix "refs/tags/" src.rev}/checkpoint/CHANGELOG.md";
```

Comment (GaetanLepage) [reply pkgs/development/python-modules/orbax-checkpoint/default.nix]: This ended up being harder than I thought X)

File: pkgs/development/python-modules/orbax-checkpoint/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/orbax-checkpoint/default.nix:1]: We could remove `importlib-resources` and `jaxlib` from dependencies.
Besides this LGTM.

**Review by natsukium — APPROVED** [review]

**Pull request merged**